Xamarin.Forms.Platform.Android/GradientStrokeDrawable.cs

```diff
@@ -0,0 +1,179 @@
+﻿using System;
+using Android.Graphics;
+using Android.Graphics.Drawables;
+using Android.Graphics.Drawables.Shapes;
+using AColor = Android.Graphics.Color;
+
+namespace Xamarin.Forms.Platform.Android
+{
+	public class GradientStrokeDrawable : PaintDrawable
+	{
+		readonly Paint _strokePaint;
+		AColor _backgroundColor;
+
+		public GradientStrokeDrawable()
+		{
+			_strokePaint = new Paint
+			{
+				Dither = true,
+				AntiAlias = true
+			};
+			_strokePaint.SetStyle(Paint.Style.Stroke);
+		}
+
+		public void SetColor(AColor backgroundColor)
+		{
+			_backgroundColor = backgroundColor;
+			InvalidateSelf();
+		}
+
+		public void SetStroke(int strokeWidth, AColor strokeColor)
+		{
+			_strokePaint.StrokeWidth = strokeWidth;
+			_strokePaint.Color = strokeColor;
+			InvalidateSelf();
+		}
+
+		public void SetGradient(Brush brush)
+		{
+			if (brush is LinearGradientBrush linearGradientBrush)
+			{
+				var p1 = linearGradientBrush.StartPoint;
+				var x1 = (float)p1.X;
+				var y1 = (float)p1.Y;
+
+				var p2 = linearGradientBrush.EndPoint;
+				var x2 = (float)p2.X;
+				var y2 = (float)p2.Y;
+
+				var gradientBrushData = linearGradientBrush.GetGradientBrushData();
+				var colors = gradientBrushData.Item1;
+				var offsets = gradientBrushData.Item2;
+
+				var linearGradientShader = new LinearGradientShader(colors, offsets, x1, y1, x2, y2);
+				SetShaderFactory(new GradientShaderFactory(linearGradientShader));
+			}
+
+			if (brush is RadialGradientBrush radialGradientBrush)
+			{
+				var center = radialGradientBrush.Center;
+				float centerX = (float)center.X;
+				float centerY = (float)center.Y;
+				float radius = (float)radialGradientBrush.Radius;
+	
+				var gradientBrushData = radialGradientBrush.GetGradientBrushData();
+				var colors = gradientBrushData.Item1;
+				var offsets = gradientBrushData.Item2;
+
+				var radialGradientShader = new RadialGradientShader(colors, offsets, centerX, centerY, radius);
+				SetShaderFactory(new GradientShaderFactory(radialGradientShader));
+			}
+		}
+
+		protected override void OnDraw(Shape shape, Canvas canvas, Paint paint)
+		{
+			base.OnDraw(shape, canvas, paint);
+
+			if (_backgroundColor != null)
+				paint.Color = _backgroundColor;
+
+			shape.Draw(canvas, _strokePaint);
+		}
+
+		public abstract class IGradientShader
```

Comment:
Is this supposed to be `interface` ? Or should it just be called a `GradientShared`?

Reply:
Changed to `GradientShader`